cli/src/semgrep/app/scans.py

```diff
@@ -48,166 +47,123 @@ class ScanCompleteResult:
 
 
 class ScanHandler:
-    def __init__(self, dry_run: bool = False, deployment_name: str = "") -> None:
-        self._deployment_id: Optional[int] = None
-        self._deployment_name: str = deployment_name
-
+    def __init__(self, dry_run: bool = False) -> None:
         self.local_id = str(uuid4())
         self.scan_metadata = out.ScanMetadata(
             cli_version=out.Version(__VERSION__),
             unique_id=out.Uuid(self.local_id),
             requested_products=[],
+            dry_run=dry_run,
         )
-        self.scan_id = None
-        self.ignore_patterns: List[str] = []
-        self._policy_names: List[str] = []
-        self._autofix = False
-        self._deepsemgrep = False
+        self.scan_response: Optional[out.ScanResponse] = None
```

Comment:
once we change ScanHandler to be built at ScanResponse time, this field will never be None and we can actually remove lots of the boilerplace accessor code below.

Reply:
agreed, see the earlier thread for more details on timing.